File: refycore/src/main/java/com/tecknobit/refycore/helpers/RefyEndpointsSet.java
package com.tecknobit.refycore.helpers;

import com.tecknobit.equinox.environment.helpers.EquinoxBaseEndpointsSet;

/**
 * The {@code RefyEndpointsSet} class is a container with all the Refy's system base endpoints
 *
 * @author N7ghtm4r3 - Tecknobit
 *
 */
public class RefyEndpointsSet extends EquinoxBaseEndpointsSet {

    /**
     * {@code UPDATE_MEMBER_ROLE_ENDPOINT} the endpoint to execute the update of a member role action
     */
    public static final String UPDATE_MEMBER_ROLE_ENDPOINT = "/updateRole";

    /**
     * {@code LEAVE_ENDPOINT} the endpoint to leave from a team
     */
    public static final String LEAVE_ENDPOINT = "/leave";

    /**
     * {@code CUSTOM_LINKS_ENDPOINT} the endpoint to fetch the custom links of a user
     */
    public static final String CUSTOM_LINKS_ENDPOINT = "/customLinks";

}

File: core/src/commonMain/kotlin/com/tecknobit/refycore/helpers/RefyEndpointsSet.kt
package com.tecknobit.refycore.helpers

import com.tecknobit.equinoxcore.network.EquinoxBaseEndpointsSet

/**
 * The `RefyEndpointsSet` class is a container with all the Refy's system base endpoints
 *
 * @author N7ghtm4r3 - Tecknobit
 */
object RefyEndpointsSet : EquinoxBaseEndpointsSet() {

    /**
     * `CHANGE_TAG_NAME_ENDPOINT` the endpoint to change the tag name of the user
     */
    const val CHANGE_TAG_NAME_ENDPOINT: String = "/changeTagName"

    /**
     * `CHANGE_MEMBER_ROLE_ENDPOINT` the endpoint to change the role of a member
     */
    const val CHANGE_MEMBER_ROLE_ENDPOINT: String = "/changeRole"

    /**
     * `LEAVE_ENDPOINT` the endpoint to leave from a team
     */
    const val LEAVE_ENDPOINT: String = "/leave"

    /**
     * `CUSTOM_LINKS_ENDPOINT` the endpoint to fetch the custom links of a user
     */
    const val CUSTOM_LINKS_ENDPOINT: String = "/customLinks"

}
